packages/cli/src/cmds/agentInstaller/agentProcedure.ts

```diff
@@ -108,6 +106,151 @@ export default abstract class AgentProcedure {
     return validationResult;
   }
 
+  async filesLocallyModified(): Promise<string[]> {
+    let files: string[] = [];
+
+    let gitStatus;
+    try {
+      const stdout = runSync(new CommandStruct('git', ['status', '-s'], this.installer.path));
+      if (stdout.length > 0) {
+        const lines = stdout.split('\n');
+        for (const line of lines) {
+          gitStatus = line.substring(0, 2);
+          const file = line.substring(3);
+          if (gitStatus[0] !== '?') {
+            files.push(file);
+          }
+        }
+      } else {
+        gitStatus = '[no status]';
+      }
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
```

Comment:
It's possible for this to catch non-Git-related errors. So it can be better to `try...catch` just the `runSync` command and then leave the rest of the code unguarded.

Reply:
True.  Changed, here and in the other functions that use `runSync`.